third_party/move/move-stdlib/src/natives/bcs.rs

```diff
@@ -63,9 +63,11 @@ fn native_to_bytes(
     };
     // serialize value
     let val = ref_to_val.read_ref()?;
+
+    let function_value_extension = context.function_value_extension();
     let serialized_value = match ValueSerDeContext::new()
         .with_legacy_signer()
-        .with_func_args_deserialization(context.function_value_extension())
+        .with_func_args_deserialization(&function_value_extension)
```

Comment:
Not really clear why its cleaner to do this in a let but fine ;-)

Reply:
agreed, I'm fine either way. Are we maybe expecting the logic to get function_value_extension to change?

Reply:
Before, context would capture `&Resolver` so `function_value_extension()` would return `&dyn FunctionValueExtension`. Now, as `Resolver` is gone, `function_value_extension()` returns `FunctionValueExtensionAdapter`, so I need to create a variable for it to be able to pass a live reference into (de)serialization context. Hence, lifted out `let function_value_extension = context.function_value_extension();`